RecoHI/HiJetAlgos/plugins/HiFJRhoFlowModulationProducer.cc

```diff
@@ -36,9 +38,51 @@
 #include <vector>
 
 namespace {
-  double lineFunction(double* x, double* par) { return par[0]; }
+
   double flowFunction(double* x, double* par) {
-    return par[0] * (1. + 2. * (par[1] * std::cos(2. * (x[0] - par[2])) + par[3] * std::cos(3. * (x[0] - par[4]))));
+    int nFlow = par[0];             // Number of fitted flow components is defined in the first parameter
+    double firstFittedVn = par[1];  // The first fitted flow component is defined in the second parameter
+
+    // Add each component separately to the total fit value
+    double flowModulation = par[2];
+    for (int iFlow = 0; iFlow < nFlow; iFlow++) {
+      flowModulation +=
+          par[2] * 2.0 * par[2 * iFlow + 3] * std::cos((iFlow + firstFittedVn) * (x[0] - par[2 * iFlow + 4]));
+    }
+    return flowModulation;
+  }
+
+  double weightFunction(double* x, double* par) {
+    // If the particle flow candidate is farther than 0.4 from the jet axis, no need for weighting due to area where no particle flow candidates can be found
+    if (x[0] * x[0] > 0.16)
+      return 0;
+
+    // If the particle flow candidate is closer than 0.4 in phi from the jet axis, then there is part of phi acceptance from which no particle flow candidates can be found. Calculate the half of the size of that strip in eta
+    double exclusionArea = TMath::Sqrt(0.16 - x[0] * x[0]);
+
+    // Particle flow caldidates are only considered until eta = par[0]. We should not exclude any area beyond that. Check that the exclusion are does not go beyond that
+
+    // First, check cases where the jet is found outside of the particle flow candidate acceptance in eta
+    if (TMath::Abs(x[1]) > par[0]) {
```

Comment:
I think I'm missing what x[1] is, but can it be negative?
It might make sense to define a new variable which is `TMath::Abs(x[1])`, to avoid repeated calls to Abs.  Maybe you should even define it as TMath::Abs(x[1]) - par[0]`,